roxie/ccd/ccdprotocol.cpp

```diff
@@ -1739,9 +1739,12 @@ class RoxieSocketWorker : public ProtocolQueryWorker
         unsigned remainingHttpConnectionRequests = global->maxHttpConnectionRequests ? global->maxHttpConnectionRequests : 1;
         unsigned readWait = WAIT_FOREVER;
 
+        SpanTimeStamp spanStartTimeStamp(true);
         Owned<IHpccProtocolMsgContext> msgctx = sink->createMsgContext(startTime);
 
 readAnother:
+        std::map<std::string, CCycleTimer> phaseTimers;
```

Comment:
Time to process is important here.  I would special case, rather than a general solution that uses string table hashes.
Also the code should aim to minimize the number of calls to record the time - the xml parse time is already recorded - it needs to be reused.